quantum/encoder.h

```diff
@@ -22,7 +22,20 @@
 #include "gpio.h"
 #include "util.h"
 
-#ifdef ENCODER_ENABLE
```

Comment:
Apparently removing this ifdef breaks the build for tests, because `platforms/test` does not define the `pin_t` type, and now things like `encoder_event_t queue[MAX_QUEUED_ENCODER_EVENTS];` which depend on `NUM_ENCODERS_LEFT`, are declared unconditionally.